### ISSUE TITLE
new addon: Download project from MyStuff

### ISSUE THREAD

**Issue by OneShot-Niko** [ScratchAddons/ScratchAddons#7185]
### Why this would be helpful

Often times I realize that a project I'd made won't load and instead crashes the page, outside or inside the editor.

I could finally work on this in an offline editor rather than painfully either restart from the beginning or wait awhile until i take <sub>~~the cranky's advice meme's advice and~~</sub> git a gud device

### How this addon works

Adds a download button to download the project.

### Possible alternatives

#6931 (PR)

### Additional context

doesn't exist

**Comment by WorldLanguages** [comment]
> doesn't exist for this lol

Could you write your issues a bit more seriously from now on? 😅 It's a bit distracting, just leave the field blank.

**Comment by OneShot-Niko** [comment]
> Could you write your issues a bit more seriously from now on? 😅 It's a bit distracting, just leave the field blank.

Yeah, my apologies - No idea why I even do it like this at this point

**Comment by Samq64** [comment]
Duplicate of #5968?

**Comment by WorldLanguages** [comment]
I think this is okay as a separate issue. See comment https://github.com/ScratchAddons/ScratchAddons/issues/5968#issuecomment-1847920112

**Comment by OneShot-Niko** [comment]
> I think this is okay as a separate issue. See comment [#5968 (comment)](https://github.com/ScratchAddons/ScratchAddons/issues/5968#issuecomment-1847920112)

The issue itself does mention adding it to mystuff, though:

> This addon could add a download button in My Stuff and/or project pages, potentially chosen by the user.

**Comment by mybearworld** [comment]
The only way to do this would require fetching all assets separately anyways, so I'm not sure if this would help in your use case.

**Comment by OneShot-Niko** [comment]
> The only way to do this would require fetching all assets separately anyways, so I'm not sure if this would help in your use case.

Yeah, the only thing I can think about is that as well...

**Comment by DNin01** [comment]
Be aware that Scratch's [Terms of Service](https://mitscratch.freshdesk.com/en/support/solutions/articles/4000219182-scratch-terms-of-service) don't permit data scraping (see §11.6).

**Comment by CST1229** [comment]
> Be aware that Scratch's [Terms of Service](https://mitscratch.freshdesk.com/en/support/solutions/articles/4000219182-scratch-terms-of-service) don't permit data scraping (see §11.6).

would they even care

**Comment by DNin01** [comment]
> > Be aware that Scratch's [Terms of Service](https://mitscratch.freshdesk.com/en/support/solutions/articles/4000219182-scratch-terms-of-service) don't permit data scraping (see §11.6).
> 
> would they even care

Well, it might be less of an issue if you're only "scraping" your own material.

**Comment by CST1229** [comment]
> > > Be aware that Scratch's [Terms of Service](https://mitscratch.freshdesk.com/en/support/solutions/articles/4000219182-scratch-terms-of-service) don't permit data scraping (see §11.6).
> > 
> > 
> > would they even care
> 
> Well, it might be less of an issue if you're only "scraping" your own material.

and if it's only one project (exact same thing the scratch website itself does)

**Comment by Samq64** [comment]
It's worth noting there is already a download button addon for the project page.

**Comment by DNin01** [comment]
> It's worth noting there is already a download button addon for the project page.

And it's probably OK because it interfaces with the official web client.
